### PR TITLE
[feature] log performance actions

### PR DESCRIPTION
**Pull request by csucrab**
produce a performance log for every action when a logger exists.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,6 +36,9 @@ database driver with extended features.
 const sqlx = require('sqlx')
 const client = sqlx.createClient({
   connection_timeout: 1000, // destroy connection on timeout
+  logger: logFunction,      // if a log function is provided,
```
Comment (yinrong) [inline README.md]: logAction: ...

File: lib/Connection.js [hunk 2]
```diff
@@ -33,6 +33,23 @@ module.exports = Connection
 
 const $ = require('./$')
 
+function logPerformance(logger, start_time, table, method, args) {
```
Comment (yinrong) [inline lib/Connection.js]: logActionInternal

File: lib/Connection.js [hunk 3]
```diff
@@ -33,6 +33,23 @@ module.exports = Connection
...
+  }
+
+  const end_time = +new Date()
+  const cost_time = end_time - start_time
+  log_obj = {
+    table: table,
+    method: method,
+    cost: cost_time,
+    // remove callback function
+    args: $._.pickBy(args, x => {return !$.util.isFunction(x)}),
```
Comment (yinrong) [inline lib/Connection.js]:
```
x => {return !$.util.isFunction(x)}
```
can be
```
x => !$.util.isFunction(x)
```

File: lib/createClient.js [hunk 4]
```diff
@@ -35,6 +35,7 @@ Client.prototype.define = function(table, config) {
 Client.prototype.getConnection = function(options) {
   var opts = $._.clone(options)
   opts.connection_timeout = this._opts.connection_timeout || DEFAULT_TIMEOUT
+  opts.logger = this._opts.logger
```
Comment (yinrong) [inline lib/createClient.js]: opts.logAction

File: README.md (hunk 1 above)
Comment (yinrong) [inline README.md]: other types of logging may be introduced in the future, and they seem have different parameters

**Comment by csucrab** [comment]
修改过了，没找到能修改assignee的地方

**Comment by yinrong** [comment]
merged

**Pull request closed without merging**